app/import_functions.go

```diff
@@ -68,6 +68,16 @@ func (a *App) ImportScheme(data *SchemeImportData, dryRun bool) *model.AppError
 		if err := a.ImportRole(data.DefaultTeamUserRole, dryRun, true); err != nil {
 			return err
 		}
+
+		if data.DefaultTeamGuestRole == nil {
+			data.DefaultTeamGuestRole = &RoleImportData{
+				DisplayName: model.NewString("Team Guest Role for Scheme"),
```

Comment:
Will this result in the guest role having no permissions? I wonder if it should instead take the default permissions from the default scheme?

Reply:
Actually is quite complex how it behaves. I create or update the schema from the import, if I'm creating it, it will create the new role for guests, and assign it to the `DefaultTeamGuestRole`, so I use that to update the "Name" of the role, and finally, during the import role, I get that role name from the database, and update the rest of the data. The unique problem that isn't handle here, is when I'm updating an already existing scheme which hasn't been migrated. Which means, you are running an old version, and trying to import a newer version of the export (which doesn't matter because we are not executing this code anyway).

Reply:
OK, I think I'm comfortable with that.